### ISSUE TITLE
Make three colored circles a bit smaller and with more spacing

### ISSUE THREAD

**Issue by mindfulsource** [lwouis/alt-tab-macos#516]
Small GUI suggestion:

I'm sure they can't be identical to macOS own design, but I suggest to make the three coloured buttons a little smaller (in relation to the window thumbnail size) and more padding between them and the thumbnail edge. Too big and a little ugly right now. They should be a little subtler.

:)

**Comment by lwouis** [comment]
Hey @mindfulsource! Thanks for sharing this feedback!

Actually I agree that they would be easier to understand if they were "identical to macOS own design". Here is some background on [how they are made currently](https://github.com/lwouis/alt-tab-macos/issues/9#issuecomment-662252405), and why they are not 1-for-1 with the system icons.

I went with this for simplicity, but as an improvement, it makes sense to improve on these icons. A worry here would be that they scale with font size, as the current implementation does. That kind of rules out using images.

Does anybody know if they exist some copyleft license vector images of macOS standard icons?

Also ideally in scope for this ticket is adding a mouse-hover and mouse-down behavior to these 3 icons.

**Comment by mindfulsource** [comment]
I'm sure SVG image format works for perfectly / infinitely granular scalability with font sizes, so could code it to be a bit smaller / more spacing in a relative (%, em) way?

If you wanted to use identical to macOS' thee dots as the graphic and also avoid potential copyright infringement, here's fa un thought, but I'm not sure if it would bring unnecessary performance hit (and/or be unnecessarily complicated to code up):

Somehow make use of the existing macOS SVG icons already in the system - you never copy them, just access them.

I'm trying to find them in Apple documentation. Here's maybe something usable, same colours:

https://developer.apple.com/design/human-interface-guidelines/macos/icons-and-images/system-icons/#status-icons

You could hunt down the real three dots in macOS file system for sure. Whether they're SVG or PNG sets though is another question.

IP law can be extremely complex. Thousands of Github projects violate lots of copyright, I'm sure. So depends how worried you are. E.g., exactly how you 'use' those images (e.g. copy and cache them in memory via AltTab software so that performance remains snappy), is one such wrinkle. IANAL though :). If you're met with a DMCA you could just go back to using non-Apple images.

Also use of such images is maybe violative of some meaningful agreement you have with Apple about how you use or re-use their IP. But if your FOSS project has no such agreement with Apple, it's fine. Maybe only a worry to you if you are also an Apple dev, where they could privately bully you as a result in retaliation.

Way simpler I think to just use something similar but not identical, which takes us back to my OP request. :) But those are some thoughts about using Apple's exact icons.

**Comment by lwouis** [comment]
**IP**

I'm not worried about this. I think Apple is fine with apps using icons from their OS, since they make that an option in the APIs. That's not a concern.

**SVG**

macOS has no native support for SVG. This means I would need a way to convert icons from SVG to bitmap images either at compile-time, or run-time. Currently, I do this at compile-time since I want original assets to exist as SVG for scalability purpose, and to be future-proof when we need to support higher resolutions.

Given how ubiquitous these icons are, there must be some SVG version of them available on some website somewhere. Anybody aware of such link?

**Comment by mindfulsource** [comment]
First I found an interesting pure CSS code reproduction of them which is cool:

https://codepen.io/atdrago/pen/yezrBR
https://gist.github.com/merqurio/4e17987b8515d44141e5952c55591869

But this is what you're after:

https://www.deviantart.com/synetcon/art/OSX-Yosemite-window-buttons-459868391

To download you have to log into a (free) DeviantArt account. I downloaded a copy of the ZIP, it looks perfect. EPS files of all hover and non-hover states of the images including useful 'no focus' grey versions, all in vector EPS form.

I note the icons changed since Yosemite (from the older skeuomorphic ones) so AFAIK these are exact as current (but it may change again in Big Sur I haven't checked).

I can upload them in SVG format if it helps?

**Comment by lwouis** [comment]
Yes please share SVG assets. This is the format of choice for this project as it is non-proprietary.

**Comment by mindfulsource** [comment]
SVG assets in a ZIP:

[osx_yosemite_window_buttons_by_synetcon_to_svg_by_mindfulsource.zip](https://github.com/lwouis/alt-tab-macos/files/5061646/osx_yosemite_window_buttons_by_synetcon_to_svg_by_mindfulsource.zip)

For reference, here's the ZIP of original EPS and PNG files:

[osx_yosemite_window_buttons_by_synetcon_d7lskpz.zip](https://github.com/lwouis/alt-tab-macos/files/5061599/osx_yosemite_window_buttons_by_synetcon_d7lskpz.zip)

License of these files: Creative Commons Attribution 3.0 License (https://creativecommons.org/licenses/by/3.0/).
Credit: synetcon at https://www.deviantart.com/synetcon/art/OSX-Yosemite-window-buttons-459868391

**Comment by lwouis** [comment]
Thanks for sharing all the assets!

I checked them, and it's not 1-for-1 I believe. I notice that the colors are not exactly like on my system (macOS 10.15.5 (19F101)). Also the pressed icons are incorrect on a few regards, but that's not an issue since I think I can just obscure the hover ones (I think that's what macOS does).

It's still an improvement compared to the current ones, that for sure.

**Comment by mindfulsource** [comment]
Oh, you're right, didn't notice that! This is trivial for me to modify in vector software, let me come back and try to produce exact copies.

If they change again when Big Sur comes out, I'm happy to update them yet again if you want. Now that I see it all, it's not hard for me to create from scratch.

(And the IP legal considerations don't really apply to me in my position...)

**Comment by mindfulsource** [comment]
Aww, I made my own little repo with a v1.0 ZIP release file (or SVGs in source):

https://github.com/mindfulsource/macos-traffic-light-buttons-as-SVG

Indeed, the pressed state ones were quite wrong. I fixed them. I also cropped the images to size and tried to make them perfectly aligned. There were slight imperfections.

---

As I note in my README, there are two other symbols I noticed: a '+' on green button for maximise vs. make full-screen, e.g. by pressing alt while mouse in green button. if you want to emulate that macOS feature in the thumbnail, I can provide graphic for that action's two visual states and update my repo. Then secondly on red button an open window whose file has unsaved changes, has a dot in the middle.

---

Hope this helps, can't wait to see them in the app if poss!

Since you won't let me donate to you, the least I can do it donate some of my time to show the love. :)

**Comment by lwouis** [comment]
Noteworthy: Apple has [Sketch/Photoshop files available here](https://developer.apple.com/design/resources/). It does not contain a hovered version of the window controls though

**Comment by mindfulsource** [comment]
OK. Do you want me to try and convert from those files to SVG to potentially improve some of the graphics? If so, I opened those Sketch files but didn't see any closing, minimizing, and zooming icons. What file and exactly where did you notice non-hover icons?

Even I think a visual approximation is good enough BTW, as that's all people notice anyway. But I understand the drive to be perfect.

**Comment by lwouis** [comment]
@mindfulsource I said "It does not contain a hovered version". It doesn't contain it. Your git repo does, but their downloadable assets don't. My point was that their assets are lacking. I'll use yours 👍

**Comment by lwouis** [comment]
I investigated conversion of SVG to PNG. For documentation sakes, here are 3 versions in 24x24 of the close button icon.

| Software | PNG output |
|----------|-------------|
| ImageMagick | ![close-imagemagik](https://user-images.githubusercontent.com/106195/91306847-1c36be00-e7e8-11ea-976c-40f30ab21b98.png) |
| Inkscape | ![close-inkscape](https://user-images.githubusercontent.com/106195/91306851-1d67eb00-e7e8-11ea-889d-195e58a9efca.png) |
| Sketch | ![close-sketch](https://user-images.githubusercontent.com/106195/91306853-1e008180-e7e8-11ea-9930-36ccfd8d50d8.png) |

I'm going to go with Inkspace as I think it has the sharpest results

**Comment by mindfulsource** [comment]
ah got it, cool. Yeah Inkscape ones looks the sharpest.

**Comment by lwouis** [comment]
I implemented this. It will be included in the next release. @mindfulsource note that I made a [separate commit](https://github.com/lwouis/alt-tab-macos/pull/553/commits/bc8887e1977f884f9bef6c21e0812c763d6dafea) with you as the author, to mark your contribution in the commit history. This will add you to the list of contributors in the app, the website, and on github.

Also note that I [corrected the colors a bit in another commit](https://github.com/lwouis/alt-tab-macos/pull/553/commits/d92971aaba94ff637d47c08fa407e4add63022e6). You may want to cherry-pick those changes in your git repo.

**Comment by mindfulsource** [comment]
I'll still investigate your color comment, but because I may deprioritise my macOS project activity, here's a good lead for finding macOS icons in future, this Big Sur Linux theme has SVGs: https://www.opendesktop.org/p/1405530/ (many macOS themes there.)

**Comment by mindfulsource** [comment]
Finally checked this @lwouis, and you're right. I thought I turned off all colour-modifying apps when measuring but I guess I didn't. Oops. I've updated my repo with a note at the top linking to your comment but I won't take the time to re-measure and update it myself just now due to deprioritising macOS related contribution compared to Linux. Hopefully Linux themes linked above have it even better.
